server/proxy/list.go

```diff
@@ -21,19 +21,42 @@ func (p *Proxy) GetInstalledApp(r *incoming.Request, appID apps.AppID) (*apps.Ap
 	return p.store.App.Get(r, appID)
 }
 
-func (p *Proxy) GetInstalledApps(r *incoming.Request) []apps.App {
-	installed := p.store.App.AsMap(r)
-	out := []apps.App{}
-	for _, app := range installed {
-		out = append(out, app)
+func (p *Proxy) GetInstalledApps(r *incoming.Request, ping bool) (installed []apps.App, reachable map[apps.AppID]bool) {
+	all := p.store.App.AsMap(r)
+
+	// all ping requests must respond, unreachable respond with "".
+	reachableCh := make(chan apps.AppID)
+	for _, app := range all {
+		rr, cancel := p.timeoutRequest(r, pingAppTimeout)
```

Comment:
Can we utilize this timeout logic when fetching bindings from all apps?

Reply:
Sure, when the time for it comes? I am not sure we want to be timing out more aggressively than the default, yet, since the binding fetch should be in theory async to the user, right?